login/service.go

```diff
@@ -86,46 +89,23 @@ func (gh *gitHubOAuth) Perform(ctx *app.AuthorizeLoginContext) error {
 			return ctx.TemporaryRedirect()
 		}
 
-		emails, err := gh.getUserEmails(ctx, ghtoken)
-		fmt.Println(emails)
-
-		primaryEmail := filterPrimaryEmail(emails)
-		if primaryEmail == "" {
-			fmt.Println("No primary email found?! ", emails)
-			ctx.ResponseData.Header().Set("Location", knownReferer+"?error="+PrimaryEmailNotFoundError)
-			return ctx.TemporaryRedirect()
-		}
-		users, err := gh.users.Query(account.UserByEmails([]string{primaryEmail}), account.UserWithIdentity())
+		identity, identityErr := gh.getUserIdentity(ctx, ghtoken)
 		if err != nil {
+			switch identityErr.code {
+			case noPrimaryEmail:
+			case associatedUserNotFound:
+				ctx.ResponseData.Header().Set("Location", knownReferer+"?error="+identityErr.message.(string))
+				return ctx.TemporaryRedirect()
+			case unauthorizedGitHubRequest:
+				return ctx.Unauthorized(identityErr.message.(*app.JSONAPIErrors))
+			}
 			ctx.ResponseData.Header().Set("Location", knownReferer+"?error=Associated user not found "+err.Error())
```

Comment:
Instead of using the hardcoded string `Associated user not found` you could use the string *constant* you defined above.